### PR TITLE
Disable `Create column` button while the column name is empty 

### LINKED ISSUE #25116: After the name of the new kanban in the projects, press Enter will not create the kanban
### Description

In a projects click new column button to input name, after press Enter key then new column is not created

### Gitea Version

1.20.0+dev-1003-g58536093b

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

command-line

### Database

None

### PR DESCRIPTION
**Pull request by puni9869**
Jun-10-2023 18-43-04

Fixes #25116

### CODE REVIEW

**Comment by delvh** [comment]
I have to say, without your GIF I wouldn't have had a single idea what this PR is fixing or what the original issue wanted…

**Comment by wxiaoguang** [comment]
There are some nits in code.

Before reviewing, the question is how does this fix #25116 ? #25116 said "click new column button to input name", then the input field is not empty?

**Comment by puni9869** [comment]
You are right I tried creating a kanban board using Enter key It is  working fine.

My bad  I miss interpret the issue.
We can change the title of the issue.
'Disable Create column button while the column name is empty'

This pr is addressed the issue in the Pr.

File: web_src/js/features/repo-projects.js [hunk 1]
```diff
@@ -9,6 +9,20 @@ function updateIssueCount(cards) {
   parent.getElementsByClassName('board-card-cnt')[0].textContent = cnt;
 }
 
+function createNewBoard(boardTitle, projectColorInput) {
+  $.ajax({
+    url: $(this).data('url'),
```
Comment (wxiaoguang) [inline web_src/js/features/repo-projects.js]: What does `this` point to? I guess it is wrong.

File: templates/repo/projects/view.tmpl [hunk 2]
```diff
@@ -33,8 +33,8 @@
 							</div>
 
 							<div class="text right actions">
-								<button class="ui cancel button">{{$.locale.Tr "settings.cancel"}}</button>
-								<button data-url="{{$.RepoLink}}/projects/{{$.Project.ID}}" class="ui primary button" id="new_board_submit">{{$.locale.Tr "repo.projects.column.new_submit"}}</button>
+								<button class="ui cancel button board-cancel-button">{{$.locale.Tr "settings.cancel"}}</button>
```
Comment (wxiaoguang) [inline templates/repo/projects/view.tmpl]: `board-cancel-button` is never used?

File: templates/repo/projects/view.tmpl [hunk 3]
```diff
@@ -33,8 +33,8 @@
...
+								<button data-url="{{$.RepoLink}}/projects/{{$.Project.ID}}" class="ui primary button disabled new-board-create" id="new_board_submit">{{$.locale.Tr "repo.projects.column.new_submit"}}</button>
```
Comment (wxiaoguang) [inline templates/repo/projects/view.tmpl]: `new-board-create` is never used?

File: web_src/js/features/repo-projects.js [hunk 4]
```diff
@@ -168,24 +182,27 @@ export function initRepoProject() {
...
   });
 
-  $('#new_board_submit').on('click', function (e) {
+  $('#new_board_submit').on('click', (e) => {
     e.preventDefault();
-
     const boardTitle = $('#new_board');
     const projectColorInput = $('#new_board_color_picker');
+    if (boardTitle.val().length < 1) {
+      return false;
```
Comment (wxiaoguang) [inline web_src/js/features/repo-projects.js]: There is already `preventDefault` above, so here can be a simple `return`

And the check can be simplified to `    if (!boardTitle.val()) {`

File: web_src/js/features/repo-projects.js [hunk 5]
```diff
@@ -168,24 +182,27 @@ export function initRepoProject() {
...
-      headers: {
-        'X-Csrf-Token': csrfToken,
-      },
-      contentType: 'application/json',
-      method: 'POST',
-    }).done(() => {
-      boardTitle.closest('form').removeClass('dirty');
-      window.location.reload();
-    });
+  $('.new-board').on('keyup', (e) => {
```
Comment (wxiaoguang) [inline web_src/js/features/repo-projects.js]: It's better to use `input` instead.

File: web_src/js/features/repo-projects.js [hunk 6]
```diff
@@ -168,24 +182,27 @@ export function initRepoProject() {
...
+    const boardTitle = $('#new_board');
+    const projectColorInput = $('#new_board_color_picker');
+    if (boardTitle.val().length < 1) {
+      $('#new_board_submit').addClass('disabled');
+      return false;
```
Comment (wxiaoguang) [inline web_src/js/features/repo-projects.js]: Do not `return false` in keyboard event listener.

File: web_src/js/features/repo-projects.js [hunk 7]
```diff
@@ -168,24 +182,27 @@ export function initRepoProject() {
...
+    }
+    $(`#new_board_submit`).removeClass('disabled');
+    if (e.key === 'Enter' || e.keyCode === 13) {
```
Comment (wxiaoguang) [inline web_src/js/features/repo-projects.js]: Why check both? Using the modern property is enough

File: web_src/js/features/repo-projects.js (hunk 5 above)
Comment (puni9869) [reply web_src/js/features/repo-projects.js]: Are you referring to this  `$('<selector>').input((e) => {})`.

I didn't find any api for this event in jquery docs https://api.jquery.com/category/events/
https://api.jquery.com/category/events/keyboard-events/

Please guide me.

File: web_src/js/features/repo-projects.js (hunk 1 above)
Comment (puni9869) [reply web_src/js/features/repo-projects.js]: Fixed

File: web_src/js/features/repo-projects.js (hunk 6 above)
Comment (puni9869) [reply web_src/js/features/repo-projects.js]: I am not using e.preventDefault in `key` event handler. Shall I.

File: web_src/js/features/repo-projects.js (hunk 5 above)
Comment (wxiaoguang) [reply web_src/js/features/repo-projects.js]: `input` is the modern event, you do: `on('input', ...)`

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event

For example, right click, paste doesn't trigger 'keyup' but it triggers 'input'.

File: web_src/js/features/repo-projects.js (hunk 6 above)
Comment (wxiaoguang) [reply web_src/js/features/repo-projects.js]: No need to prevent anything in input event listener.

File: web_src/js/features/repo-projects.js [hunk 8]
```diff
@@ -168,24 +183,29 @@ export function initRepoProject() {
...
-      headers: {
-        'X-Csrf-Token': csrfToken,
-      },
-      contentType: 'application/json',
-      method: 'POST',
-    }).done(() => {
-      boardTitle.closest('form').removeClass('dirty');
-      window.location.reload();
-    });
+  $('.new-board').on('input keyup', (e) => {
```
Comment (puni9869) [inline web_src/js/features/repo-projects.js:197]: I have added `input` event. But only binding input event will not capture the `e.key` `Enter` we need to use keyup as well.

Comment (wxiaoguang) [reply web_src/js/features/repo-projects.js:197]: You need 2 separate event listeners, one for "input", one for "keyup enter"

**Review by wxiaoguang — APPROVED** [review]
Although there are nits, overall LGTM.

**Review by lunny — APPROVED** [review]

**Pull request merged**